math/exp.c

```diff
@@ -23,9 +23,18 @@ double frexp(double x, int *exp)
 {
 	conv_t *conv = (conv_t *)&x;
 
+	if (isnan(x) != 0) {
+		return NAN;
+	}
+
+	if (isinf(x) != 0) {
+		return x;
+	}
+
 	*exp = 0;
 
 	if ((x == 0.0) || (x == -0.0)) {
+		*exp = 0;
```

Comment:
As I have written, `*exp = 0` can be moved before this `if` - now, if `*exp = 0` is also inside `if`, `*exp = 0` is executed twice if the condition holds true. It should be removed from this line.